.github/docker/rez-win-py/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# escape=`
+
+# Tag of base image to use
+ARG BASE_TAG
+
+# Base image windows version
+FROM rez-win-base:$BASE_TAG
+
+# NOTE: Any " requires \" in the Dockerfile for windows.
+# NOTE: The order matters. ARG after the shell command will allow access via
+# the PowerShell environment like ${ENV:PYTHON_VERSION}.
+SHELL ["powershell.exe", "-NoLogo", "-NoProfile", "-ExecutionPolicy", "ByPass"]
+
+# Python version to install (full chocolatey compatible version required)
+# For example: 3.7.5
+#
+ARG PYTHON_VERSION
+
+# ------------------------------------------------------------------------------------------------------------
+# Python
+#
+# Installs given Python version to C:\Python
+# Python 2.x uses msi while 3.x has an exe with separate arguments
+#
+RUN ${PYTHON_INSTALL_PATH} = 'C:\Python';                                                                                            `
```

Comment:
I'm curious, what are the floating back quotes for?

Reply:
Essentially it is just like a backslash for multi-line commands.
Here's the deal with quoting on windows based Dockerfiles. First you should define this:
```
#escape=`
```
https://docs.docker.com/engine/reference/builder/#escape

This way you can use backslash in script as expected by PowerShell for paths and backticks for line ends with multi-line commands or quoting.

FYI backtick is the native escape character in Powershell since backslash is (wrongly) reserved for paths.

You still need to escape `\"` for `"` which you need when you want to expand a variable, while in most cases you can get away with a single `'` when using static strings.

One also has to be careful because variable expansion changes in docker AFTER the `shell` like, but in my best attempts there is no way to use a variable in the `FROM` statement with PowerShell convention.

Hence only this schema works:
```
ARG ONE
FROM $ONE<- bash-like expansion
SHELL powershell
ARG TWO <- Powershell expansion
RUN ${ENV:TWO}
```
While this will just drive you mad:
```
SHELL powershell
ARG ONE
FROM $ONE<- NO
FROM ${ENV:ONE}<- NO
What happened to my ARG value? ¯\_(ツ)_/¯
```
So the lesson, to my surprise, is that the commands may not necessarily affect global state.